src/controllers/plugins/api/Message.cpp

```diff
@@ -163,7 +162,60 @@ std::unique_ptr<MessageElement> elementFromTable(const sol::table &tbl)
     assert(el);
 
     el->setTrailingSpace(tbl.get_or("trailing_space", true));
-    el->setTooltip(tbl.get_or("tooltip", QString{}));
+
+    auto link = tbl.get<sol::optional<Link>>("link");
+    if (link)
+    {
+        bool ok = false;
+        el->setLink(*link, &ok);
+        if (!ok)
+        {
+            throw std::runtime_error("'link' not supported on type='" +
+                                     type.toStdString() + '\'');
+        }
+        QString tooltip = tbl.get_or("tooltip", QString{});
+        if (!tooltip.isEmpty())
+        {
+            tooltip += "<br>";
+        }
+
+        switch (link->type)
+        {
+            case Link::Url:
+                tooltip += QString("<b>URL:</b> %1").arg(link->value);
+                break;
+            case Link::UserAction:
+                tooltip += QString("<b>Command:</b> %1").arg(link->value);
+                break;
+            case Link::CopyToClipboard:
+                tooltip +=
+                    QString("<b>Copy to clipboard:</b> %1").arg(link->value);
```

Comment:
If they will be in there, the values should probably be trimmed (or even left out?).

- For Url: Trimming is probably the best here. Though, one could still hide malicious stuff in the end.
- For UserAction: Trim. We should also make sure that we only ever send one message, so `hello\n/unban PluginAuthor` won't send as two messages.
- For CopyToClipboard: Maybe hiding (i.e. only show that this copies _something_)? There's not really much harm you can do other than to overwrite the clipboard. Not sure about all use-cases, but I could imagine something like "click to copy token" or something. You wouldn't want that to show.